docs/codeql/codeql-language-guides/extensible-predicates.rst

```diff
@@ -0,0 +1,182 @@
+.. _extensible-predicates:
+
+:orphan:
+:nosearch:
+
+Extensible predicates and their interaction with data extensions
+================================================================
+
+You can use data extensions to model the methods and callables that control dataflow in any framework or library. This is especially useful for custom frameworks or niche libraries, that are not supported by the standard CodeQL libraries.
+
+.. include:: ../reusables/beta-note-model-packs-java.rst
+
+About this article
+------------------
+
+This reference article describes the available inputs for the extensible predicates, including access paths, kinds, and provenance.
+
+Sources, sinks, summaries, and neutrals are commonly known as models. These models support several shared arguments and a few model-specific arguments. The arguments populate a series of columns for each extensible predicate.
+
+About extensible predicates
+---------------------------
+
+At a high level, there are two main components to using data extensions. The query writer defines one or more extensible predicates in their query libraries. CLI and code scanning users who want to augment these predicates supply one or more extension files whose data gets injected into the extensible predicate during evaluation. The extension files are either stored directly in the repository where the codebase to be analyzed is hosted, or downloaded as CodeQL model packs.
+
+This example of an extensible predicate for a source is taken from the core Java libraries https://github.com/github/codeql/blob/main/java/ql/lib/semmle/code/java/dataflow/ExternalFlowExtensions.qll#L8-L11
+
+.. code-block:: ql
+
+    extensible predicate sourceModel(
+      string package, string type, boolean subtypes, string name, 
+      string signature, string ext, string output, string kind, 
+      string provenance
+    );
+
+An extensible predicate is a CodeQL predicate with the following restrictions:
+
+- It uses the ``extensible`` keyword.
+- It has no body.
+- All predicate parameters have primitive types.
+- It is not in a module.
+
+Columns shared by all extensible predicates
```

Comment:
This is specific to the extensible predicates for Models as Data ("all" in this context means the extensible predicates describes in `data-extensions-to-model-java-dependencies` and should not included in the general description of extensible predicates.

Reply:
The same goes for the remaining content in this file.

Reply:
Comment added to internal issue for follow up.